client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/GetWatchResponse.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.lucene.uid.Versions;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class GetWatchResponse {
+    private final String id;
+    private final long version;
+    private final WatchStatus status;
+
+    private final BytesReference source;
+    private final XContentType xContentType;
+
+    public GetWatchResponse(String id) {
+        this(id, Versions.NOT_FOUND, null, null, null);
+    }
+
+    public GetWatchResponse(String id, long version, WatchStatus status, BytesReference source, XContentType xContentType) {
+        this.id = id;
+        this.version = version;
+        this.status = status;
+        this.source = source;
+        this.xContentType = xContentType;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public long getVersion() {
+        return version;
+    }
+
+    public boolean isFound() {
+        return version != Versions.NOT_FOUND;
+    }
+
+    public WatchStatus getStatus() {
+        return status;
+    }
+
+    public BytesReference getSource() {
+        return source;
+    }
+
+    public XContentType getContentType() {
```

Comment:
I wonder if it would make sense to always return JSON content as we do in `GetFieldMappingsResponse`. In that class, it also looks like we have a `sourceAsMap()` for convenience.